File: OOP_Variable/3.rb
class Calc  #대문자로 시작
  #생성자
  #메소드 생성
  def initialize(v1,v2)
    #p v1, v2
    @v1 = v1  # @v1: 인스턴스 변수
    @v2 = v2
  end
  def add()
    return @v1+@v2
  end
  def sub()
    return @v1-@v2
  end
  def setV1(v)
    if v.is_a?(Integer) # v가 Integer인지?
      @v1 = v
    end
  end
  def getV1()
    return @v1
  end
end

#객체 사용
c1 = Calc.new(10,10)
p c1.add()
p c1.sub()
# c1.setV1(20)
c1.setV1('one')
p c1.add()

File: OOP_Variable/3.py
class Calc(object):
    def __init__(self,v1,v2): # 첫번째 입력값이 v1, 두번째 입력값이 v2
        # print(v1, v2)
        if isinstance(v1, int):
            self.v1 = v1    #인스턴스 변수
        if isinstance(v2, int):
            self.v2 = v2
    def add(self):  #관습적으로 self라고 함
        return self.v1+self.v2
    def sub(self):
        return self.v1-self.v2

    def setV1(self, v):
        if isinstance(v, int):
            self.v1 = v
    def getV1(self):
        return self.v1
#객체 사용
c1 = Calc(10,10)
print(c1.add())
print(c1.sub())

c1.setV1('one')
c1.v2 = 30
print(c1.add())

return
c2 = Calc(30,20)
print(c2.add())
print(c2.sub())
